### PR TITLE
Moving RecoveryServices module from Hydra to Swagger

### PR DESCRIPTION
**Pull request by dragonfly91**
## Description

Moved RecoveryServices module from Hydra to Swagger based client.
---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by markcowl** [comment]
@dragonfly91 Build failures, please take a look

**Comment by dragonfly91** [comment]
Fixed build failures. Please take a look!

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/RemoveAzureRMRecoveryServicesVault.cs [hunk 1]
```diff
@@ -44,11 +44,11 @@ public override void ExecuteCmdlet()
         {
             try
             {
-                RecoveryServicesOperationStatusResponse response = RecoveryServicesClient.DeleteVault(this.Vault.ResourceGroupName, this.Vault.Name);
+                var response = RecoveryServicesClient.DeleteVault(this.Vault.ResourceGroupName, this.Vault.Name);
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/RemoveAzureRMRecoveryServicesVault.cs]: Also, cmdlet should support ShouldProcess, and if you do output something, it should be identified in the OutputType attribute

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs [hunk 2]
```diff
@@ -50,11 +50,9 @@ public override void ExecuteCmdlet()
             {
                 if (this.BackupStorageRedundancy.HasValue)
                 {
-                    UpdateVaultStorageTypeRequest vaultStorageRequest = new UpdateVaultStorageTypeRequest();
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs]: Also, per the previous cmdlet, this needs to support SHouldProcess, and when you update the output, add an OutputType attribute with the appropriate OutputType

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/NewAzureRMRecoveryServicesVault.cs [hunk 3]
```diff
@@ -56,13 +56,13 @@ public override void ExecuteCmdlet()
         {
             try
             {
-                VaultCreateArgs vaultCreateArgs = new VaultCreateArgs();
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/NewAzureRMRecoveryServicesVault.cs]: Need to support ShouldProcess in this cmdlet, and need to specify the OutputType using an attribute

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRmRecoveryServicesBackupProperties.cs [hunk 4]
```diff
@@ -45,10 +45,10 @@ public override void ExecuteCmdlet()
         {
             try
             {
-                GetResourceStorageConfigResponse getStorageResponse = RecoveryServicesClient.GetVaultStorageType(
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRmRecoveryServicesBackupProperties.cs]: Name of thsi cmdlet should be singular ```Get-AzureRmBackupValutProperty```

**Comment by cormacpayne** [comment]
@dragonfly91 Hey Samuel, would you mind pulling the latest changes from the `preview` branch? These changes removed the dll-Help.xml file from the repository. The markdown help files are now used to generate the help and were added to the csproj file, causing your PR to have merge conflicts.

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Common/ClientRequestIdHandler.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Commands.RecoveryServices
+{
+    /// <summary>
+    /// Delegating handler to add the client request id header to the request.
+    /// </summary>
+    public class ClientRequestIdHandler : DelegatingHandler, ICloneable
```
Comment (dragonfly91) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Common/ClientRequestIdHandler.cs]: This is only for our cmdlets.

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Utilities/Logger.cs [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            writeDebugAction = writeDebug;
+            writeVerboseAction = writeVerbose;
+            throwTerminatingErrorAction = throwTerminatingError;
+        }
+
+        /// <summary>
+        /// Writes verbose message.
+        /// </summary>
+        /// <param name="text"></param>
+        public void WriteVerbose(string text)
```
Comment (dragonfly91) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Utilities/Logger.cs]: Sure, added this in the XML comment.

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/ChangeLog.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    * Overview of change #3
+    * Overview of change #4
+        - Additional information about change #4
+
+    ## YYYY.MM.DD - Version X.Y.Z (Previous Release)
+    * Overview of change #1
+        - Additional information about change #1
+-->
+## Current Release
+* Moved client from Hydra to Swagger
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/ChangeLog.md]: Hydra/Swagger don't mean anything to customers, please describe the actual interface changes to cmdlets (new cmdlets, new parameters, new output, shouldprcess support)

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Common/PSRecoveryServicesClient.cs [hunk 8]
```diff
@@ -69,49 +80,38 @@ public RecoveryServicesManagementClient GetRecoveryServicesClient
...
+
+        private ResourceManagementClient resourceManagementClient;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="PSRecoveryServicesClient" /> class with 
         /// required current subscription.
         /// </summary>
         /// <param name="azureSubscription">Azure Subscription</param>
-        public PSRecoveryServicesClient(IAzureProfile azureProfile)
+        public PSRecoveryServicesClient(IAzureProfile azureProfile, AzureContext defaultContext)
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Common/PSRecoveryServicesClient.cs]: Why do you need the profile?

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRmRecoveryServicesBackupProperties.cs (hunk 4 above)
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRmRecoveryServicesBackupProperties.cs]: This change is still required

**Comment by dragonfly91** [comment]
Hi @markcowl The checks are passing now. Can you please merge the PR?

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1492/

Filed an issue here to improve the changelog entries: https://github.com/Azure/azure-powershell/issues/3868

**Pull request merged**